aspects/schema.go

```diff
@@ -794,3 +845,19 @@ func (v *arraySchema) parseConstraints(constraints map[string]json.RawMessage) e
 }
 
 func (v *arraySchema) expectsConstraints() bool { return true }
+
+type validationError struct {
+	path    []string
```

Comment:
maybe for later but I wonder this should use an interface instead of strings so that we can deal better to avoid things like:

a.[0].foo  vs  a[0].foo ?

Reply:
Sure, I can improve this as a follow-up. I've added a task